0000-experiment-dataframe.md

```diff
@@ -0,0 +1,320 @@
+
+
+#### CurveAnalysisArtifact
+
+This artifact have two extra fields.
+
+- fit_status: A dictionary (that may provide nice html view) of fitting status. This is currently provided as analysis result entry with the name `@Parameters_*`. This is not really the analysis results and thus moved to artifact. This contains information about fitting, such as fit models, solver status, initial and final fit parameters and so forth. For example,
+
+- curve_data: Curve data points in the data frame format. This table may have the column:
+  - x_val (float): X value.
+  - y_val (float): Y value.
+  - y_err (float): Y standard error.
+  - samples (int): Number of samples used to evaluate this y value and error.
+  - model (str): Name of fit model associated with this data.
+  - group (str): Name of fit group if composite curve analysis, e.g. Hamiltonian tomography analysis.
+  - data_kind (str): Data type. Typically "raw" or "formatted". Raw data is used for visualization.
+
+
+### C. Experiment service
+
+The experiment service acts as an intermediate layer between QE
+and the resultDB. The current structure of an analysis result contains the following fields:
+
+- uuid (str): Same as the proposed `result_id` field.
+- experiment_uuid (str): Same as the proposed `experiment_id` field.
+- type (str): Same as the proposed `name` field.
+- chisq (float): Same as the proposed `chisq` field.
+- device_components (List[DeviceComponent]): Same as the proposed `device_components` field.
+- quality (str): One of three possible values: `Bad`, `No Information`, `Good`. This is slightly different than the strings being used in QE.
+- tags (List[str]): Same as the proposed `tags` field.
+- verified (bool): Same as the proposed `verified` field.
+- device_name (str): Same as the proposed `backend` field.
+- created_at (string): An ISO 8601 timestamp indicating the creation time of the analysis
+result inside the resultDB. This is similar to the `creation_datetime` field but not the same.
+Usually experimentalists are interested in the time at which the experiment was done, rather than a timestamp for the resultDB entry.
+This suggests these time information must be handled separately.
+- fit (dict): A free-form dictionary allowing the storage of arbitrary data.
+This can be used to store the `value`, `extra` and `source` fields.
+
+Two of the proposed fields do not make sense in a resultDB context and won't be saved/loaded but only handled by QE:
+- `created_in_db`: This has been conventionally managed internally in the `ExperimentData`.
+- `experiment`: This may be stored in the `fit` field. Not only experiment UUID, but also human readable representation of experiment is important to help experimentalists with filtering and sorting analysis parameters.
+
+To reduce the amount of work done by QE, we suggest that the experiment service `qiskit-ibm-experiment` will implement
```

Comment:
I would like if the experiment service has its classes and functionality to work with saving/loading, and updating/modifying (where allowed) raw data, analysis results, artifacts (which I think could include figures), and that experiments is basically just initializing and populating the values of these classes to be saved, or modifying loaded ones with as little special handling has possible.

Reply:
I agree, the question is how the minimal interface `qiskit-experiments` uses would look like; I can make the changes in the `qiskit-ibm-experiment` to accommodate for (hopefully) anything we'd like.